yt_dlp/extractor/cspan.py

```diff
@@ -242,3 +243,49 @@ def get_text_attr(d, attr):
                 'title': title,
                 'id': 'c' + video_id if video_type == 'clip' else video_id,
             }
+
+
+class CSpanCongressIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?c-span\.org/congress/'
+    IE_DESC = 'C-SPAN'
+    _TESTS = [{
+        'url': 'https://www.c-span.org/congress/?chamber=house&date=2017-12-13&t=1513208380',
+        'info_dict': {
+            'id': 'Congressional Chronicle - Members of Congress, Hearings and More | C-SPAN.org',
+            'title': 'Congressional Chronicle - Members of Congress, Hearings and More',
+            'description': 'md5:54c264b7a8f219937987610243305a84',
+            'ext': 'mp4'
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._generic_id(url)
+        webpage = self._download_webpage(url, display_id)
+        file = open("sample.txt", "w")
+        file.write(webpage)
+        file.close()
+        video_url = re.findall(r'sources: \[\{file: \'(.*?)\'\}\],', webpage)[0]
+        formats = self._extract_m3u8_formats(video_url, display_id, ext='mp4')
+        self._sort_formats(formats)
+
+        basic_url = "/".join(url.rstrip('/').split('/')[:3])
+        for f in formats:
+            f.setdefault('http_headers', {})['referer'] = basic_url + "/"
+            f.setdefault('http_headers', {})['origin'] = basic_url
+            f.setdefault('http_headers', {})['accept'] = "*/*"
+            f.setdefault('http_headers', {})['sec-fetch-dest'] = "empty"
+            f.setdefault('http_headers', {})['sec-fetch-mode'] = "cors"
+            f.setdefault('http_headers', {})['sec-fetch-site'] = "cross-site"
```

Comment:
```suggestion
            f.setdefault('http_headers', {}).update({
                'Referer': basic_url + '/',
                'Origin': basic_url,
                'Accept': '*/*',
                'Sec-Fetch-Dest': 'empty',
                'Sec-Fetch-Mode': 'cors',
                'Sec-Fetch-Site': 'cross-site'
            })
```

Reply:
Most of these headers are probably unnecessary too

Reply:
Also u can just hardcode the `basic_url` to be `'https://www.c-span.org'`
And i'm pretty sure having `/` in the end for origin aswell won't break things so u can have the basic_url as 
`'https://www.c-span.org'/`

Reply:
I don't know whether this website does, but sometimes `Origin` header is strictly checked

Reply:
```suggestion
            f.setdefault('http_headers', {})['Referer'] = 'https://www.c-span.org/'
```
Not directly tested, but seems like only `Referer` is actually needed